adam-core/src/main/scala/org/bdgenomics/adam/converters/VariantContextConverter.scala

```diff
@@ -412,148 +336,844 @@ private[adam] class VariantContextConverter(dict: Option[SequenceDictionary] = N
+                                        idx: Int,
+                                        indices: Array[Int]): VariantCallingAnnotations.Builder = {
+    // see https://github.com/samtools/htsjdk/issues/741
+    val gtFiltersWereApplied = true
+    if (gtFiltersWereApplied) {
+      val filtersWereApplied = vcab.setFiltersApplied(true)
+      if (g.isFiltered) {
+        filtersWereApplied.setFiltersPassed(false)
+          .setFiltersFailed(g.getFilters.split(";").toList)
+      } else {
+        filtersWereApplied.setFiltersPassed(true)
+      }
+    } else {
+      vcab.setFiltersApplied(false)
+    }
+  }
+
+  private[converters] def formatFisherStrandBias(g: HtsjdkGenotype,
+                                                 vcab: VariantCallingAnnotations.Builder,
+                                                 idx: Int,
+                                                 indices: Array[Int]): VariantCallingAnnotations.Builder = {
+    Option(g.getExtendedAttribute("FS"))
+      .map(attr => {
+        tryAndCatchStringCast(attr, attribute => {
+          vcab.setFisherStrandBiasPValue(attribute.asInstanceOf[java.lang.Float])
+        }, attribute => {
+          vcab.setFisherStrandBiasPValue(attribute.toFloat)
+        })
+      }).getOrElse(vcab)
+  }
+
+  private[converters] def formatRmsMapQ(g: HtsjdkGenotype,
+                                        vcab: VariantCallingAnnotations.Builder,
+                                        idx: Int,
+                                        indices: Array[Int]): VariantCallingAnnotations.Builder = {
+    Option(g.getExtendedAttribute("MQ"))
+      .map(attr => {
+        tryAndCatchStringCast(attr, attribute => {
+          vcab.setRmsMapQ(attribute.asInstanceOf[java.lang.Float])
+        }, attribute => {
+          vcab.setRmsMapQ(attribute.toFloat)
+        })
+      }).getOrElse(vcab)
+  }
+
+  private[converters] def formatMapQ0(g: HtsjdkGenotype,
+                                      vcab: VariantCallingAnnotations.Builder,
+                                      idx: Int,
+                                      indices: Array[Int]): VariantCallingAnnotations.Builder = {
+    Option(g.getExtendedAttribute("MQ0"))
+      .map(attr => {
+        tryAndCatchStringCast(attr, attribute => {
+          vcab.setMapq0Reads(attribute.asInstanceOf[java.lang.Integer])
+        }, attribute => {
+          vcab.setMapq0Reads(attribute.toInt)
+        })
+      }).getOrElse(vcab)
+  }
+
+  private val annotationFormatFieldConversionFns: Iterable[(HtsjdkGenotype, VariantCallingAnnotations.Builder, Int, Array[Int]) => VariantCallingAnnotations.Builder] = Iterable(
+    formatFilters(_, _, _, _),
+    formatFisherStrandBias(_, _, _, _),
+    formatRmsMapQ(_, _, _, _),
+    formatMapQ0(_, _, _, _)
+  )
+
+  private[converters] def extractFilters(vca: VariantCallingAnnotations,
+                                         gb: GenotypeBuilder): GenotypeBuilder = {
+    Option(vca.getFiltersApplied)
```

Comment:
This is probably correct, but is much harder to read than the non-functional way it is written for Variant filters.  Sorry, I'm getting old.  :)